lib/types/gpt.nix

```diff
@@ -195,24 +210,31 @@ in
         if ! blkid "${config.device}" >&2; then
           sgdisk --clear "${config.device}"
         fi
-        ${lib.concatStrings (map (partition: ''
-          # try to create the partition, if it fails, try to change the type and name
-          if ! sgdisk \
-            --align-end ${lib.optionalString (partition.alignment != 0) ''--set-alignment=${builtins.toString partition.alignment}''} \
-            --new=${toString partition._index}:${partition.start}:${partition.end} \
-            --change-name="${toString partition._index}:${partition.label}" \
-            --typecode=${toString partition._index}:${partition.type} \
-            "${config.device}"
-          then sgdisk \
-            --change-name="${toString partition._index}:${partition.label}" \
-            --typecode=${toString partition._index}:${partition.type} \
-            "${config.device}"
-          fi
-          # ensure /dev/disk/by-path/..-partN exists before continuing
-          partprobe "${config.device}" || : # sometimes partprobe fails, but the partitions are still up2date
-          udevadm trigger --subsystem-match=block
-          udevadm settle
-        '') sortedPartitions)}
+        ${lib.concatStrings (map (partition:
+          let
+            args = ''
+              --partition-guid="${toString partition._index}:${if partition.uuid == null then "R" else partition.uuid}" \
+              --change-name="${toString partition._index}:${partition.label}" \
+              --typecode=${toString partition._index}:${partition.type} \
+              "${config.device}" \
+            '';
+            createArgs = ''
+              --align-end ${lib.optionalString (partition.alignment != 0) ''--set-alignment=${builtins.toString partition.alignment}''} \
+              --new=${toString partition._index}:${partition.start}:${partition.end} \
+            '';
```

Comment:
not sure why this was split, but isn' the device more of a createArg?

Reply:
Just to avoid repeating itself, nothing more. It seemed nicer this way. I can revert it to the original style if you want.